playgrounds/eav-checkbox-playground.R

```diff
@@ -0,0 +1,559 @@
+rm(list = ls(all = TRUE)) #Clear the memory of variables from previous run. This is not called by knitr, because it's above the first chunk.
+
+# ---- load-sources ------------------------------------------------------------
+
+# ---- load-packages -----------------------------------------------------------
+library("magrittr")
+requireNamespace("dplyr")
+requireNamespace("readr")
+requireNamespace("testit")
+
+# ---- declare-globals ---------------------------------------------------------
+# as projects are only containing test data, okay to commit URL and tokens
+# to Github in private repo
+# remove if we ever submit a PR to the main project
+redcap_uri <- "https://redcap.hbcd.cmig-ucsd.org/api/"
+
+# token      <- "53320377CD34F60392870F5F6C20953B" # longitudinal & repeating
+# token      <- "778FAF179F3DBEDBC1A3CA1CAB2B36EF" # repeating
+# token      <- "831D17A42D03BC448A7B1A08CDE19AD9" # longitudinal
+token      <- "B77827EB144D9B7D74F01E932C9F0B29" # neither
+
+fields  <- NULL
+# fields   <- c(
+#   # "record_id",
+#   "form_1_check_001"
+# )
+forms   <- NULL
+# forms   <- c(
+#   "form_1",
+#   # "form_2",
+#   "form_3"
+#   # "form_4"
+# )
+events  <- NULL
+# events  <- c(
+#   "event_1_arm_1"
+#   # "event_3_arm_1"
+# )
+records <- NULL
+
+blank_for_gray_form_status <- TRUE
+.complete_value_for_untouched_forms <- dplyr::if_else(
+  blank_for_gray_form_status,
+  NA_character_,
+  as.character(REDCapR::constant("form_incomplete"))
+)
+
+# NOTE: I think that an additional function parameter to specify how to handle
+#   untouched checkbox items might make sense. But we can also make it dependent
+#   on `blank_for_gray_form_status` if you prefer that. I just think some people
+#   might want to make different selections for either.
+blank_for_untouched_checkbox <- TRUE
+
+# ---- load-data ---------------------------------------------------------------
+# View(REDCapR::redcap_variables(redcap_uri, token)$data)
+system.time({
+  meta <- REDCapR:::redcap_metadata_internal(
+    redcap_uri,
+    token
+  )
+
+  ds_metadata          <- meta$d_variable
+  ds_metadata_plumbing <- ds_metadata |>
```

Comment:
What's the meaning of plumbing in this context, and what does it do?  You seem never use it again

Reply:
That's how the main author of `REDCapR` calls variables that are standard variables that REDCap governs (vs. user-specified variables). This includes ID, event, repeating instance, etc. fields

Reply:
If it's not used below, it's probably just a remainder of code that the `REDCapR` author wrote and that I started from. We can delete if not needed